src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/PublicPrivateDataCombiner.java

```diff
@@ -0,0 +1,136 @@
+package uk.gov.companieshouse.overseasentitiesapi.utils;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang3.tuple.Pair;
+import uk.gov.companieshouse.api.model.beneficialowner.PrivateBoDataApi;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.model.managingofficerdata.ManagingOfficerDataApi;
+import uk.gov.companieshouse.api.model.officers.CompanyOfficerApi;
+import uk.gov.companieshouse.api.model.psc.PscApi;
+import uk.gov.companieshouse.api.model.update.OverseasEntityDataApi;
+import uk.gov.companieshouse.overseasentitiesapi.service.PrivateDataRetrievalService;
+import uk.gov.companieshouse.overseasentitiesapi.service.PublicDataRetrievalService;
+
+public class PublicPrivateDataCombiner {
+
+  private final PublicDataRetrievalService publicDataRetrievalService;
+  private final PrivateDataRetrievalService privateDataRetrievalService;
+  private final String salt;
+
+  public PublicPrivateDataCombiner(final PublicDataRetrievalService publicDataRetrievalService,
```

Comment:
Any pair in the map which only has private data is a consistency error between the apis. I wonder should we check for this case and print errors and remove them from the map. A pcs/officer having only public data is also a consistency error (we should warn) but we have no choice but to work with that. The combining code is going to have to be very careful with this incomplete pair though!

Reply:
I think we need to take the error checking into a new story. While it might be more efficient to dedect missing public/private BO/MOs there are other cases we also need to consider.